mk/spksrc.checksum.mk

```diff
@@ -39,42 +39,47 @@ pre_checksum_target: checksum_msg
 
 # validate file integrity with the provided digests.
 checksum_target: $(PRE_CHECKSUM_TARGET)
-	@if [ -f $(DIGESTS_FILE) ] ; \
-	then \
-	  cat $(DIGESTS_FILE) | (cd $(DISTRIB_DIR) && while read file type value ; \
-	  do \
-	    if [ $$file != $(LOCAL_FILE) ] ; \
-	    then \
-	      $(MSG) "  Downloaded file $(LOCAL_FILE) not in digests file" ; \
-	      rm $(DOWNLOAD_COOKIE) ; \
-	      exit 1 ; \
-	    fi ; \
-	    if [ ! -f $$file ] ; \
-	    then \
-	      $(MSG) "  File $$file not downloaded" ; \
-	      rm $(DOWNLOAD_COOKIE) ; \
-	      exit 1 ; \
-	    fi ; \
-	    case $$type in \
-	      SHA1|sha1)     tool=sha1sum ;; \
-	      SHA256|sha256) tool=sha256sum ;; \
-	      MD5|md5)       tool=md5sum ;; \
-	      *) $(MSG) "Unsupported digest type $$type" ; exit 1 ;; \
-	    esac ; \
-	    $(MSG) "  Checking $$tool of file $$file"; \
-	    if echo "$$value  $$file" | $$tool --status -c - ; \
-	    then \
-	      true ; \
-	    else  \
-	      $(MSG) "    Wrong $$tool for file $$file" ; \
-	      [ -f $$file.wrong ] && rm $$file.wrong ; \
-	      mv $$file $$file.wrong ; \
-	      $(MSG) "    Renamed as $$file.wrong" ; \
-	      rm $(DOWNLOAD_COOKIE) ; \
-	      $(MSG) "    Download cookie removed to trigger the download again" ; \
-	      exit 1 ; \
-	    fi \
-	  done) ; \
+	@if [ -f $(DIGESTS_FILE) ] ; then \
+	  # validate file in digests \
```

Comment:
Last suggestion, as all the `exit` cases are on top, this one could be using a similar approach and benefit from a `exit 1` like the others afterwards:
```
@if [ ! -f $(DIGESTS_FILE) ] ; then \
	  $(MSG) "No digests file for $(NAME)" ; \
	  exit 1 ; \
else \
```

Reply:
This is not implemented with `exit 1` now and I doubt that we should change this.
The only package I found without digests file is cross/squidclamav. I wanted to add one, but the current source package is too old and not available for download anymore.